docs/faq.md

```diff
@@ -0,0 +1,157 @@
+  when the analysis is *complex*: other tools are frequently limited
+  in these dimensions
+
+- when data analysis is *too expensive*.  Sometimes you *can* use a
+  very expensive parallel engine to analyze large data volumes, but
+  the analysis would be much cheaper using F, since it is using many
+  fewer resources.
+
+## How is the F engine programmed?
+
+F is programmed in SQL.  F supports essentially the entire SQL
+language.  There exists a low-level Rust API as well, but F's tooling
+is geared towards the SQL API.
+
+## How is F different from streaming engines?
+
+Existing streaming engines offer a subset of F's capabilities.
+
+Like a database, F supports arbitrary insertions and deletions for
+arbitrary data sources.  Many streaming engines only support
+insertions.
+
+F is programmed using the SQL database language.
+
+## How does F differ from databases?
+
+Database engines are not very efficient at maintaining multiple
+complex views on fast-changing data.
+
+## How can you write streaming programs in SQL?
+
+The SQL programs use SQL *tables* to describe data sources, and SQL
+*views* to describe the computed results.  Then, as the input tables
+are modified, F guarantees that the views are always up-to-date.
+
+## How fast is F?
+
+This depends on a lot of factors, but for typical queries F can
+compute the results in a matter of milliseconds after each change.
+
+## Why is F fast?
+
+F uses a technique called "Incremental View Maintenance".  The idea is
+to compute only as much as necessary based on the *changes* of the
+inputs.  Often the changes are much smaller than the whole input.
+This means that computing based on the changes can be even millions of
+times faster than computing the result from scratch.
+
+## Don't databases offer incremental view maintenance?
+
+Databases can only evaluate some restricted types of views
+incrementally.  F can do this for essentially any program you can
+write in SQL.
+
+## Is F a database?
+
+No.  F only offers the data processing engine, but does not
+store the data persistently for later retrieval.
+
+## Where does F run?
+
+F is run as a cloud service.  The part that does data processing
+(the dataplane) is designed to be deployed in the customers'
+datacenters, as dictated by requirements of the data sources and
+privacy.
+
+## Where is the data coming from?
+
+A F program can ingest data from any number and variety of data
+sources.
+
+Data can be sent to F using http requests, or F can use connectors to
+retrieve data from a variety of sources (e.g., Kafka, databases, data
+lakes, etc.).  If there is no connector for your favorite data sources
+we can usually prototype one in a few days.
+
+## Where is the output?
+
+F is designed to work as a service which notifies users of changes to
+all the views they defined, whenever these changes happen.
+
+## I want the views, not the changes.
+
+F can do that too, but it may be a bit slower, especially when the
```

Comment:
We don't really have this yet (we should!), and when we do, I don't think that we should be discouraging users from downloading a complete view if that's what they need.  We can point out that the best practice is to download the full snapshot and after that only stream changes.

Reply:
I'd say that you stream the changes into a database and query the database.

Reply:
We already offer the complete output views that one can enumerate. So no need for the database.